appengine/pond/duck/rank/js/rank.js

```diff
@@ -16,44 +16,84 @@
  */
 
 /**
- * @fileoverview Creates a view of all the ducks for a user.
- * @author aschmiedt@google.com (Abby Schmiedt)
+ * @fileoverview Computes leaderboard ranking matches.
  */
 'use strict';
 
 goog.provide('Pond.Duck.Rank');
 
+goog.require('Pond.Avatar');
+goog.require('Pond.Battle');
+goog.require('Pond.Duck');
 goog.require('Pond.Duck.Datastore');
 
+Pond.Duck.Rank.MATCH_KEY = null;
+Pond.Duck.Rank.RANK_ENTRIES = [];
+
 /**
- * .
+ * Initialize interpreter and triggers match request. Called on page load.
  */
 Pond.Duck.Rank.init = function () {
-  // Start background match request computation.
-  Pond.Duck.Datastore.getMatchRequest(Pond.Duck.Rank.computeRankRequest);
+  Pond.Battle.isHeadless = true;
+  Pond.Battle.GAME_FPS = 1000;
+  Pond.Battle.TIME_LIMIT = 60 * 1000;
+
+  // Lazy-load the JavaScript interpreter.
+  BlocklyInterface.importInterpreter();
+
+  // Start match request computation.
+  Pond.Duck.Datastore.getMatchRequest(Pond.Duck.Rank.handleMatchRequest);
+};
+
+/**
+ * Computes ranking match.
+ */
+Pond.Duck.Rank.computeMatch = function() {
+  if (!('Interpreter' in window)) {
+    // Interpreter lazy loads and hasn't arrived yet.  Try again later.
+    setTimeout(Pond.Duck.Rank.computeMatch, 250);
+    return;
+  }
+  Pond.Battle.reset();
```

Comment:
Since Battle is a singleton, is this going to have troubles when a user is running a match and this gets called.

Reply:
I didn't notice any issues when I ran it, but I'll double check. The reason I think this works is because the rank battle is run in a separate iframe so I think they are different instances.

Reply:
I set breakpoints and confirmed that the battle run using the "run program" button is unaffected by background matches (the Battle properties like isHeadless are correct).